spec/factories/book_ranks.rb

```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :book_rank do
+    book { nil }
+    rank { nil }
```

Comment:
Since it is not nullable
```suggestion
    book { create(:book) }
    rank { create(:rank) }
```